File: 0657-robot-return-to-origin/Solution.java
// Return true if the robot returns to the origin after it finishes all of its moves

class Solution {
    public boolean judgeCircle(String moves) {
        int r = 0;
        int l = 0;
        int u = 0;
        int d = 0;

        for (char m : moves.toCharArray()) {
            if (m == 'R') { r++; }
            else if (m == 'L') { l++; }
            else if (m == 'U') { u++; }
            else { d++; }
        }

        return (r == l) && (u == d);
    }
}

File: 0657-robot-return-to-origin/Solution.js
// Return true if the robot returns to the origin after it finishes all of its moves

/**
 * @param {string} moves
 * @return {boolean}
 */

var judgeCircle = function (moves) {
    let arr = moves.split("");
    let horiz = 0;
    let vert = 0;

    arr.forEach((e) => {
        if (e == 'L') { horiz++; }
        else if (e == 'R') { horiz--; }
        else if (e == 'U') { vert++; }
        else { vert--; }
    });

    return horiz == 0 && vert == 0;
};
